equinox_models/RNN.py

```diff
@@ -0,0 +1,105 @@
+import abc
+
+import equinox as eqx
+import jax
+import jax.numpy as jnp
+
+
+class _AbstractRNNCell(eqx.Module):
+    """Abstract RNN Cell class."""
+
+    cell: eqx.Module
+    hidden_size: int
```

Comment:
The intention for this was so you don't need to pass hidden_dim to the RNN instance as you already have by creating the cell, seems I forgot this later and only used it once in RNN's __init__, modified it now so that hidden_dim doesn't need to be passed to RNN and is inferred from the cell you pass